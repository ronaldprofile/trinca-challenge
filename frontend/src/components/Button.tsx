import { styled } from "../../stitches.config";

export const Button = styled("button", {
  height: 50,
  padding: "0 20px",
  borderRadius: 4,
  fontSize: 16,
  fontWeight: 700,
  textTransform: "uppercase",

  variants: {
    shape: {
      primary: {
        background: "#FFD836",
        color: "White",
        border: "2px solid #FFD836",
        transition: ".2s all",

        "&:hover": {
          filter: "brightness(0.85)",
        },

        "&:focus": {
          outline: "none",
        },

        "&:disabled": {
          cursor: "not-allowed",
          filter: "brightness(0.75)",
        },
      },

      secondary: {
        background: "Black",
        color: "White",
        border: "1px solid Black",
        transition: ".2s all",
      },

      danger: {
        background: "#dc2626",
        color: "White",
        border: "1px solid #dc2626",
        transition: ".2s all",

        "&:hover": {
          filter: "brightness(0.80)",
        },
      },
    },

    color: {
      green: {
        background: "#34CB79",
      },
    },
  },

  defaultVariants: {
    shape: "primary",
  },
});
