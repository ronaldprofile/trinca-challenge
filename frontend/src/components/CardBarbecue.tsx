import { Link } from "react-router-dom";
import { Users, CurrencyCircleDollar } from "phosphor-react";
import { formatPrice } from "../utils/formatCurrency";
import { formatDate } from "../utils/formatDate";
import Skeleton from "react-loading-skeleton";
import { IBarbecue } from "../types";

interface CardBarbecueProps {
  barbecue: IBarbecue;
  isFetchingBarbecue: boolean;
}

export function CardBarbecue({
  barbecue,
  isFetchingBarbecue,
}: CardBarbecueProps) {
  const dateFormatted = formatDate(new Date(barbecue.scheduled_day), {
    day: "2-digit",
    month: "2-digit",
  });

  const dateToBarbecueFormattedFull = formatDate(
    new Date(barbecue.scheduled_day),
    {
      day: "numeric",
      month: "numeric",
      year: "numeric",
    }
  );

  return (
    <Link
      to={`/details/${barbecue.id}`}
      title="ver detalhes"
      className={`w-full p-6 bg-white text-black shadow-lg rounded-sm
        border-2 border-transparent hover:-translate-y-2 transition-all
      `}
    >
      <div className={`flex flex-col gap-2`}>
        {!isFetchingBarbecue ? (
          <time
            dateTime={new Date(barbecue.scheduled_day).toISOString()}
            title={dateToBarbecueFormattedFull}
            className={`font-bold text-base sm:text-[28px]`}
          >
            {dateFormatted}
          </time>
        ) : (
          <Skeleton width={84} height={33} />
        )}

        {!isFetchingBarbecue ? (
          <span className={`font-bold text-xl text-black/80`}>
            {barbecue.title}
          </span>
        ) : (
          <Skeleton width={124} height={25} />
        )}
      </div>

      <div className={`mt-12 flex items-center justify-between`}>
        {!isFetchingBarbecue ? (
          <span
            className={`text-sm sm:text-xl font-medium flex items-center gap-3`}
          >
            <Users size={24} />
            {barbecue.members.length}
          </span>
        ) : (
          <Skeleton width={84} height={25} />
        )}

        {!isFetchingBarbecue ? (
          <span
            className={`text-base sm:text-xl font-medium flex items-center gap-3`}
          >
            <CurrencyCircleDollar size={24} />
            {formatPrice(barbecue.amount_collected)}
          </span>
        ) : (
          <Skeleton width={84} height={25} />
        )}
      </div>
    </Link>
  );
}
