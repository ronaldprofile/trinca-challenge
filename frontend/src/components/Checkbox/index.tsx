import * as CheckboxPrimitive from "@radix-ui/react-checkbox";
import { StyledCheckbox } from "./styles";

type CheckboxProps = CheckboxPrimitive.CheckboxProps & {};

export function Checkbox({ children, ...props }: CheckboxProps) {
  return <StyledCheckbox {...props}>{children}</StyledCheckbox>;
}

type CheckboxIndicatorProps = CheckboxPrimitive.CheckboxIndicatorProps & {};

export function CheckboxIndicator({ children }: CheckboxIndicatorProps) {
  return (
    <CheckboxPrimitive.CheckboxIndicator>
      {children}
    </CheckboxPrimitive.CheckboxIndicator>
  );
}
