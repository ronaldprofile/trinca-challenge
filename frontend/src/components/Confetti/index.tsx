import { useState } from "react";
import ReactConfetti from "react-confetti";
import { useWindowSize } from "react-use";

export function Confetti() {
  const { width, height } = useWindowSize();

  return (
    <ReactConfetti
      width={width}
      height={height}
      recycle={false}
      colors={["#8257e5", "#04d361"]}
      numberOfPieces={600}
      tweenDuration={6000}
    />
  );
}
