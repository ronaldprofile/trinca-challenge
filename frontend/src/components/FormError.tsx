import { keyframes, styled } from "../../stitches.config";

const animationFade = keyframes({
  from: {
    transform: "translateY(-14px)",
    opacity: 0,
    visibility: "hidden",
  },

  to: {
    transform: "translateY(0)",
    opacity: 1,
    visibility: "visible",
  },
});

const InputError = styled("div", {
  paddingTop: 7,

  animation: `${animationFade} 800ms`,
  span: {
    fontSize: 12,
    color: "#d34242",
  },
});

interface FormErrorProps {
  error?: string;
}


export function FormError({ error }: FormErrorProps) {
  if (!error) {
    return <></>;
  }

  return (
    <InputError>
      <span>{error}</span>
    </InputError>
  );
}
