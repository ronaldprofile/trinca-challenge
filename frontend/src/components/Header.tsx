import React from "react";
import clsx from "clsx";

interface HeaderProps {
  title?: string;
  className?: string;
}

export function Header({ title, className }: HeaderProps) {
  return (
    <header className={clsx(`w-screen`, className)}>
      <div className={`flex items-center justify-center`}>
        <h1 className={`text-[32px] text-black font-bold`}>
          {title ? title : "Agenda de eventos"}
        </h1>
      </div>
    </header>
  );
}
