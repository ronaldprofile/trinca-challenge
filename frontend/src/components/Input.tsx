import { styled } from "../../stitches.config";

export const Input = styled("input", {
  height: 50,
  padding: "0 20px",
  fontSize: 16,
  transition: ".2s all",
  borderRadius: 4,

  variants: {
    shape: {
      primary: {
        background: "White",
        border: "1px solid transparent",

        "&:focus": {
          borderColor: "#fb1",
          outline: "none",
          boxShadow: "0 0 0 3px #fea",
        },

        "&:hover": {
          borderColor: "#fb1",
          boxShadow: "0 0 0 3px #fea",
        },
      },

      secondary: {
        background: "#E1FAEC",
        border: "1px solid #d1d5db",

        "&:focus": {
          outline: "none",
        },
      },
    },
  },

  defaultVariants: {
    shape: "primary",
  },
});
