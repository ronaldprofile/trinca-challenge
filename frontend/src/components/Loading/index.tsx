import { LoadingContainer } from "./styles";

export function Loading() {
  return (
    <LoadingContainer>
      <span></span>
      <span></span>
      <span></span>
      <span></span>
    </LoadingContainer>
  );
}
