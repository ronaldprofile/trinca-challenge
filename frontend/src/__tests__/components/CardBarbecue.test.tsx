import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { CardBarbecue } from "../../components/CardBarbecue";

const barbecue = {
  id: "1",
  title: "FIFA 23",
  description: "Jogar videogame",
  amount_collected: 300,
  scheduled_day: "10/22/2022",

  members: [
    {
      id: "1",
      name: "Ronald",
      paid: true,
      contribution: 200,
      hasDrinkIncluded: false,
    },
    {
      id: "2",
      name: "Marc",
      paid: true,
      contribution: 100,
      hasDrinkIncluded: false,
    },
  ],
};

describe("Component Card barbecue", () => {
  it("the component snapshot must always be the same", () => {
    const { container } = render(
      <CardBarbecue barbecue={barbecue} isFetchingBarbecue={false} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render title event", () => {
    render(<CardBarbecue barbecue={barbecue} isFetchingBarbecue={false} />);

    const titleEvent = screen.getByText("FIFA 23");

    expect(titleEvent).toBeInTheDocument();
  });

  it("should render schedule day event", () => {
    render(<CardBarbecue barbecue={barbecue} isFetchingBarbecue={false} />);

    const scheduleDayEvent = screen.getByText("22/10");

    expect(scheduleDayEvent).toBeInTheDocument();
  });

  it("should render total members event", () => {
    render(<CardBarbecue barbecue={barbecue} isFetchingBarbecue={false} />);

    expect(barbecue.members).toHaveLength(2);

    const totalMembers = screen.getByTestId("total_members_event");

    expect(totalMembers).toBeInTheDocument();
  });

  it("should render amount collected event", () => {
    render(<CardBarbecue barbecue={barbecue} isFetchingBarbecue={false} />);

    const amountCollectedEvent = screen.getByTestId("amount_collected_event");

    expect(amountCollectedEvent).toBeInTheDocument();

    expect(amountCollectedEvent).toHaveTextContent("R$ 300,00");
  });

  it("should navigate to see details event", async () => {
    render(<CardBarbecue barbecue={barbecue} isFetchingBarbecue={false} />);

    const linkElement = screen.getByRole("link", {
      hidden: true,
    });

    expect(linkElement).toBeInTheDocument();

    expect(linkElement).toHaveAttribute("href", "/details/1");

    userEvent.click(linkElement);

    expect(await screen.findByText("FIFA 23")).toBeInTheDocument();
  });
});
