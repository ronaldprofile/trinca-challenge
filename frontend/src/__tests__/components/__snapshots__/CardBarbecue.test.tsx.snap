// Vitest Snapshot v1

exports[`Component Card barbecue > the component snapshot must always be the same 1`] = `
<a
  class="w-full p-6 bg-white text-black shadow-lg rounded-sm
        border-2 border-transparent hover:-translate-y-2 transition-all
      "
  href="/details/1"
  title="ver detalhes"
>
  <div
    class="flex flex-col gap-2"
  >
    <time
      class="font-bold text-base sm:text-[28px]"
      datetime="2022-10-22T03:00:00.000Z"
      title="22/10/2022"
    >
      22/10
    </time>
    <span
      class="font-bold text-xl text-black/80"
    >
      FIFA 23
    </span>
  </div>
  <div
    class="mt-12 flex items-center justify-between"
  >
    <span
      class="text-sm sm:text-xl font-medium flex items-center gap-3"
      data-testid="total_members_event"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <circle
          cx="88"
          cy="108"
          fill="none"
          r="52"
          stroke="currentColor"
          stroke-miterlimit="10"
          stroke-width="16"
        />
        <path
          d="M155.4,57.9A54.5,54.5,0,0,1,169.5,56a52,52,0,0,1,0,104"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
        <path
          d="M16,197.4a88,88,0,0,1,144,0"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
        <path
          d="M169.5,160a87.9,87.9,0,0,1,72,37.4"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
      2
    </span>
    <span
      class="text-base sm:text-xl font-medium flex items-center gap-3"
      data-testid="amount_collected_event"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="128"
          x2="128"
          y1="72"
          y2="88"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="128"
          x2="128"
          y1="168"
          y2="184"
        />
        <circle
          cx="128"
          cy="128"
          fill="none"
          r="96"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
        <path
          d="M104,168h36a20,20,0,0,0,0-40H116a20,20,0,0,1,0-40h36"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
      R$Â 300,00
    </span>
  </div>
</a>
`;
